### PR TITLE
[Hotfix release-1.9] MQTT PubSub: Resume subscriptions after a reconnection

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Extracted from #2350 with the fix for the MQTT PubSub component only

The fix for the MQTT binding component will not be included in the hotfix

### CODE REVIEW

**Review by artursouza — APPROVED** [review]

File: pubsub/mqtt/mqtt.go [hunk 1]
```diff
@@ -212,34 +212,23 @@ func (m *mqttPubSub) startSubscription(ctx context.Context) error {
 		// for backwards-compatibility; see: https://github.com/dapr/components-contrib/pull/2104
 		consumerClientID += "-consumer"
 	}
-	connCtx, connCancel := context.WithTimeout(ctx, defaultWait)
-	c, err := m.connect(connCtx, consumerClientID)
-	connCancel()
+
+	readyCh := make(chan struct{})
```
Comment (mukundansundar) [inline pubsub/mqtt/mqtt.go:216]: should this be buffered ?
```suggestion
	readyCh := make(chan struct{}, 1)
```

Comment (mukundansundar) [reply pubsub/mqtt/mqtt.go:216]: @ItalyPaleAle ^

Comment (ItalyPaleAle) [reply pubsub/mqtt/mqtt.go:216]: No, it's ok for this to be unbuffered so it blocks until it's consumed

Comment (ItalyPaleAle) [reply pubsub/mqtt/mqtt.go:216]: Changed to use `close()` rather than sending a message, which won't block

**Comment by ItalyPaleAle** [comment]
I made a small change following feedback by @mukundansundar

**Comment by ItalyPaleAle** [comment]
For some cert tests that are failing, it's due to external changes in the test infra. I've opened #2356 to backport those into release-1.9

**Pull request merged**